spacer/caffe_utils.py

```diff
@@ -96,6 +99,11 @@ def load_net(modeldef_path: str,
     :param modelweighs_path: pretrained weights path.
     :return: pretrained model.
     """
+    # To verify that the correct weight is loaded
+    with open(modelweighs_path, 'rb') as fp:
```

Comment:
Hmm. So in this case we are forced to load it twice, it seems, since the loading happens in the `caffe.Net()` call. Can you measure how long this extra step takes?

Reply:
```
./usr/lib/python3.6/importlib/_bootstrap.py:219: RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 192 from C header, got 216 from PyObject
  return f(*args, **kwds)
==> time spent on checking sha: 0.44178128242492676
[libprotobuf WARNING google/protobuf/io/coded_stream.cc:604] Reading dangerously large protocol message.  If the message turns out to be larger than 2147483647 bytes, parsing will be halted for security reasons.  To increase the limit (or to disable these warnings), see CodedInputStream::SetTotalBytesLimit() in google/protobuf/io/coded_stream.h.
[libprotobuf WARNING google/protobuf/io/coded_stream.cc:81] The total number of bytes read was 552712164
==> time spent in total: 0.9515583515167236
.==> time spent in total: 8.821487426757812e-06
.==> time spent on checking sha: 0.44198107719421387
[libprotobuf WARNING google/protobuf/io/coded_stream.cc:604] Reading dangerously large protocol message.  If the message turns out to be larger than 2147483647 bytes, parsing will be halted for security reasons.  To increase the limit (or to disable these warnings), see CodedInputStream::SetTotalBytesLimit() in google/protobuf/io/coded_stream.h.
[libprotobuf WARNING google/protobuf/io/coded_stream.cc:81] The total number of bytes read was 552712164
.==> time spent in total: 1.6689300537109375e-06
..............==> time spent in total: 5.9604644775390625e-06
.==> time spent in total: 5.7220458984375e-06
..==> time spent in total: 4.291534423828125e-06
.==> time spent in total: 4.0531158447265625e-06
```
This is the measurement on the docker, roughly extra `0.44178128242492s` is spent on checking the `sha256`, check the code in `caffe_utils.py`

Reply:
@beijbom

Reply:
Ok. I think that is fine.